lib/rspec/openapi/schema_merger.rb

```diff
@@ -25,22 +25,15 @@ def normalize_keys(spec)
   # Also this needs to be aware of OpenAPI details unlike an ordinary deep_reverse_merge
   # because a Hash-like structure may be an array whose Hash elements have a key name.
   #
-  # TODO: Perform more intelligent merges like rerouting edits / merging types
   # TODO: Should we probably force-merge `summary` regardless of manual modifications?
   def deep_reverse_merge!(base, spec)
     spec.each do |key, value|
       if base[key].is_a?(Hash) && value.is_a?(Hash)
-        deep_reverse_merge!(base[key], value)
-      elsif !base.key?(key)
-        base[key] = value
-      elsif base[key].is_a?(Array) && value.is_a?(Array)
-        # parameters need to be merged as if `name` and `in` were the Hash keys.
-        if key == 'parameters'
-          base[key] |= value
-          base[key].uniq! { |param| param.slice('name', 'in') }
+        if !base[key].key?("$ref")
```

Comment:
This is a special treatment to avoid merge if the object is [Reference Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#reference-object).
The reference object can have only `$ref` field.
Invalid example:
```yaml
  schema:
     $ref: ".."
     description: "override"
```

Reply:
Is this for the case that `$ref` is manually added to an OpenAPI schema file? If you manually maintain the schema of the response, why don't you skip updating OpenAPI from such test cases? Do you have a test case that calls multiple endpoints and want to update OpenAPI for only one of these?